src/System.IO.Compression/tests/DeflateStreamTests.cs

```diff
@@ -799,7 +827,7 @@ public void Roundtrip_Write_ReadByte()
         public async Task WrapNullReturningTasksStream()
         {
             using (var ds = new DeflateStream(new BadWrappedStream(BadWrappedStream.Mode.ReturnNullTasks), CompressionMode.Decompress))
-                await Assert.ThrowsAsync<InvalidOperationException>(() => ds.ReadAsync(new byte[1024], 0, 1024));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => ds.ReadAsync(new byte[1024], 0, 1024));
```

Comment:
What caused the need for this change in exception thrown?

Reply:
The issue is that I changed DeflateStream to use the new ReadAsync overload, which returns a ValueTask.  If the target stream doesn't override the new overload, the base implementation will delegate to the existing ReadAsync overload, which returns Task, and then construct a ValueTask from that.  This bad stream returns null from ReadAsync, causing the ValueTask ctor to fail.  Technically it's a breaking change, but the stream is already in a terribly broken state if it's returning null from ReadAsync, and so which exception emerges in that case didn't seem like a reason to not use the new method.

Reply:
> Technically it's a breaking change, but the stream is already in a terribly broken state if it's returning null from ReadAsync, and so which exception emerges in that case didn't seem like a reason to not use the new method.

I don't know if these APIs have shipped/etc. (can you please give some background)... so, I am not sure if we care if this is a breaking change. I lack the context, but are there any concerns with this?

Reply:
> I don't know if these APIs have shipped

DeflateStream has been shipping for a long time.

> are there any concerns with this?

I'm not concerned.  But I welcome other's opinions... @weshaggard, @terrajobst, @JeremyKuhne?  The issue is that if someone has a Stream where they've overridden ReadAsync(byte[], ...) and are returning null from it, and don't override ReadAsync(Memory, ...), and then that Stream is wrapped in a DeflateStream, previously DeflateStream would throw an InvalidOperationException, now it ends up throwing an ArgumentNullException.  Do we care?

Reply:
I'm not concerned either- I wouldn't bend over backwards for this.